frontend/src/components/AuthUtils/index.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * Utility functions to get various parameters corresponding to the currently
+ * authenticated user. These functions should ONLY be called from private pages,
+ * i.e. the user must be authenticated to ensure the parameters can be
+ * retrieved.
+ */
+
+import { AuthUtilsConsumer, getCurrentUser } from './AuthUtilsConsumer.js';
+import { SIGN_IN } from '../../constants/routes.js';
+
+/**
+ * Called when one of the authentication utility functions are used when the
+ * user has not logged in yet. Since the utility functions require that they be
+ * called from private pages, this function being called implies that the user
+ * managed to access a private page without being authenticated first. So they
+ * are immiedately redirected to the SIGN_IN page.
+ */
+function redirectToSignIn() {
+  window.location.href = SIGN_IN;
+}
+
+export function getUserDisplayName() {
+  if (!getCurrentUser()) {
+    redirectToSignIn();
+    return null;
+  }
+  return getCurrentUser().displayName;
+}
+
+export function getUserEmail() {
+  if (!getCurrentUser()) {
+    redirectToSignIn();
+    return null;
+  }
+  return getCurrentUser().email;
+}
+
+export function getUserPhotoUrl() {
+  if (!getCurrentUser()) {
+    redirectToSignIn();
+    return null;
+  }
+  return getCurrentUser().photoURL;
+}
+
+export function getUserUid() {
```

Comment:
correct me if I'm misunderstanding this, but these three functions check if there's a user who's signed in.. right?

Reply:
I believe it checks that prior to returning the value (e.g. current user email) in case the current user is not authenticated/doesn't exist

Reply:
Ideally, these functions should only be called on private pages (where the user needs to be authenticated to access them). This guarantees that the user won't be null. If somehow the user manages to bypass PrivateRoute, then these functions check for that and will redirect them to the Sign In page.